automated install/basic-install.sh

```diff
@@ -179,6 +179,18 @@ is_command() {
     command -v "${check_command}" >/dev/null 2>&1
 }
 
+apt_package_list=
```

Comment:
If this is meant to be a global variable then it should be declare up with the rest of the variables, starting on line 28. And the name should be in all caps if it's intended to be global.

If not then it should be declared local inside the function.

Reply:
It's kind of an edge case. It is only used in the below function, but it needs to be global to allow assigning an empty string first (failsafe, so that it cannot be overwritten by an export) while the first here assigned value needs to survive for subsequent calls of this function.

I find it easier to read the code when in this case it is declared aside of the function, but I can move it to the other global functions and make it upper case, if you prefer.

Reply:
Yeah, I'd prefer all the variables to be in one place if they are not inside a function. Keeps from having to hunt down where they are. We already have the vars declared in one central place so this would become an outlier and I want to avoid things like that.

The function works from a quick test but we'd like to be more verbose with things. Use `if`s and no `||` or `&&`s. It helps with our userbase to understand.

```suggest
is_apt_package(){
    # Checks whether a package, or one that provides it, is available in
    # the installed APT repository lists.
    local check_package=$1

    # Obtain the list of available packages once
    if [ -z "$apt_package_list" ]; then
      apt_package_list=$(apt-cache dumpavail | grep -E '^P(ackage|rovides):')
    fi

    grep -qE " $check_package(,|$)" <<< "$apt_package_list"
}
```

Reply:
Okay done.